akka-diagnostics/src/main/scala/akka/diagnostics/ConfigChecker.scala

```diff
@@ -795,6 +804,47 @@ class ConfigChecker(system: ExtendedActorSystem, config: Config, reference: Conf
       else Nil
     }
 
+  private def checkRemoteDeployment(): List[ConfigWarning] =
+    ifEnabled("remote-prefer-cluster") { checkerKey =>
+      val path = "akka.actor.provider"
+      if (config.getString(path) == "remote" || config.getString(path) == "akka.remote.RemoteActorRefProvider")
+        warn(checkerKey, path, "remote is possible, but prefer cluster") ////TODO add better description?
+      else Nil
+    }
+
+  private def checkCreateActorRemotely(): List[ConfigWarning] =
+    ifEnabled("create-actor-remotely") { checkerKey =>
+      val path = """akka.actor.deployment."/...".remote""""
+
+      val isRemoteDeployment = config
+        .getConfig("""akka.actor.deployment""")
+        .withoutPath("default")
+        .entrySet()
+        .toArray()
+        .exists { case x: Entry[String, ConfigValue] =>
+          x.getKey.matches("""^\"\/.*\"\.remote""")
+        } // is of the type "/...".remote
+      if (isRemoteDeployment)
+        warn(
+          checkerKey,
+          path,
+          "Deploying an actor remotely is deprecated and not supported. As per https://doc.akka.io/docs/akka/current/remoting.html#creating-actors-remotely"
+        ) //TODO add better description?
+      else Nil
+    }
+
+  private def checkClusterWatchFailureDetector(): List[ConfigWarning] =
+    ifEnabled("cluster-watch-failure-detector") { checkerKey =>
+      val path = "akka.remote.watch-failure-detector"
+      if (isClusterConfigAvailable) {
```

Comment:
The actual check is missing here. The idea was to warn about any application.conf changes to `akka.remote.watch-failure-detector.*` properties when cluster is enabled.

That must be done with
```
config.getValue(path1) != reference.getValue(path1) || config.getValue(path2) != reference.getValue(path2)
```
Where we could pick a few properties, such as
```
val path1 = s"$path.acceptable-heartbeat-pause"
val path2 = s"$path.heartbeat-interval"
```

Reply:
I see now. The warning was about changing the defaults. That makes sense.

Reply:
added